### PR TITLE
Regenerate roughjs shape only when the item is updated

### PR DESCRIPTION
**Pull request by GasimGasimzada**
It adds back the performance optimization of regenerating the shapes only when element properties change.

Do not merge it yet! I need to check out export etc afterwards.

### CODE REVIEW

**Comment by vjeux** [comment]
Thanks for bringing it back!

File: src/index.tsx [hunk 1]
```diff
@@ -312,7 +318,11 @@ export class App extends React.Component<{}, AppState> {
         onCopy={e => {
           e.clipboardData.setData(
             "text/plain",
-            JSON.stringify(elements.filter(element => element.isSelected))
+            JSON.stringify(
+              elements
+                .filter(element => element.isSelected)
+                .map(({ shape, ...el }) => el)
```
Comment (vjeux) [inline src/index.tsx]: This is a smart way to copy without!

**Comment by vjeux** [comment]
Note that moving an element doesn't have to redraw the shape, but resizing it does. Not sure where it's being set but may be worth double checking

File: src/index.tsx (hunk 1 above)
Comment (GasimGasimzada) [reply src/index.tsx]: I removed the shape element from history, clipboard, and exported JSON file. Is there any other feature in the app that needs these exports to be removed.

**Comment by GasimGasimzada** [comment]
I am only doing it during resizing (line 683)/

**Comment by vjeux** [comment]
Perfect, thanks for checking!

**Comment by GasimGasimzada** [comment]
I am checking it now. I checked everything on existing elements and forgot to test new elements...

**Comment by GasimGasimzada** [comment]
@vjeux it should work now.

**Pull request merged**